homeassistant/components/weatherflow/__init__.py

```diff
@@ -0,0 +1,78 @@
+"""Get data from Smart Weather station via UDP."""
+from __future__ import annotations
+
+import logging
+
+from pyweatherflowudp.client import EVENT_DEVICE_DISCOVERED, WeatherFlowListener
+from pyweatherflowudp.device import EVENT_LOAD_COMPLETE, WeatherFlowDevice
+from pyweatherflowudp.errors import ListenerError
+
+from homeassistant.components.sensor import DOMAIN as SENSOR_DOMAIN
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST, EVENT_HOMEASSISTANT_STOP
+from homeassistant.core import Event, HomeAssistant, callback
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+from homeassistant.helpers.start import async_at_started
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS = [SENSOR_DOMAIN]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up WeatherFlow from a config entry."""
+
+    client = WeatherFlowListener(host=entry.data.get(CONF_HOST, "0.0.0.0"))
+    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = client
```

Comment:
We don't store data in `hass.data` until we are past the point `ConfigEntryNotReady` can be raised